modules/tensor_mechanics/plugins/elastic.f

```diff
@@ -0,0 +1,74 @@
+****************************************************************************************
+**  UMAT, FOR ABAQUS/STANDARD INCORPORATING ISOTROPIC ELASTICITY                      **
+****************************************************************************************
+****************************************************************************************
+**
+**
+**
+      SUBROUTINE UMAT(STRESS,STATEV,DDSDDE,SSE,SPD,SCD,
+     1     RPL,DDSDDT,DRPLDE,DRPLDT,
+     2     STRAN,DSTRAN,TIME,DTIME,TEMP,DTEMP,PREDEF,DPRED,CMNAME,
+     3     NDI,NSHR,NTENS,NSTATV,PROPS,NPROPS,COORDS,DROT,PNEWDT,
+     4     CELENT,DFGRD0,DFGRD1,NOEL,NPT,LAYER,KSPT,KSTEP,KINC)
+
+C
+C      INCLUDE 'ABA_PARAM.INC'
+C
+C      CHARACTER*8 CMNAME
+C
+      DIMENSION STRESS(NTENS),STATEV(NSTATV),DDSDDE(NTENS, NTENS),
+     1     DDSDDT(6),DRPLDE(6),STRAN(6),DSTRAN(NTENS),
+     2     PREDEF(1),DPRED(1),PROPS(NPROPS),COORDS(3),DROT(3,3),
+     3     DFGRD0(3,3), DFGRD1(3,3), TIME(2)
+
+      PARAMETER(ZERO=0.D0, ONE=1.D0, TWO=2.D0, THREE=3.D0)
+
+C ----------------------------------------------------------------
+C
+C     UMAT FOR ISOTROPIC ELASTICITY
+C
+C     CANNOT BE USED FOR PLANE STRESS
+C ----------------------------------------------------------------
+C
+C      PROPS(1) - E
+C
+C      PROPS(2) - NU
+C ----------------------------------------------------------------
+C
+C      IF (NDI.NE.3) THEN
+C         WRITE (7, *) ’THIS UMAT MAY ONLY BE USED FOR ELEMENTS
+C    1        WITH THREE DIRECT STRESS COMPONENTS’
+C         CALL XIT
+C      ENDIF
```

Comment:
Do we know if this should be active or removed?

Reply:
If NDI can be 2, e.g., that can be misleading.

Reply:
In MOOSE we will always have three stress components. I removed the code that set NDI based on the mesh dimension. That's not how tensor mechanics works. I'll remove all that commented out code.